hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -129,6 +129,9 @@ public static ClientMessage create() {
 
     public static ClientMessage createForEncode(int initialCapacity) {
         initialCapacity = QuickMath.nextPowerOfTwo(initialCapacity);
+        if (initialCapacity < 0) {
+            initialCapacity = Integer.MAX_VALUE;
+        }
         if (USE_UNSAFE) {
```

Comment:
@jerrinot @cangencer  The method contract explicitly limits for possible values of this method parameter but i also don't want another duplicate method for this purpose. Currently, I did put checks outside the QuickMath.nextPowerOfTwo where it is being called in two places for the protocol implementation. I still make a call with a value not matching the method contract in this fix for the sake of not duplicating the same code. Let me know if it is good enough for this fix.